shared/js/background/before-request.es6.js

```diff
@@ -274,6 +269,15 @@ function blockHandleResponse (thisTab, requestData) {
             tracker.reason = 'content blocking disabled'
         }
 
+        if (requestData.tabId === -1 && (tracker.action === 'block' || tracker.action === 'redirect')) {
```

Comment:
I'm sure you've thought about this, but it makes me nervous that this check is made later than the old one. Is there a reason you can't or don't want to keep the check in the same place? What's your thinking?

Reply:
Before we block the serviceworker initiated request, we need to know:
- The 'tab' context (initiator)
- If the URL is a tracker request we should block (taking the tracker allowlist into account)
- That there isn't an exception for the first party (user allowlist, feature allowlist etc)

This is the point in the code where we have all that information, and can make the blocking decision.

Reply:
Could we not move the check back to where it was*,  but expand it to check `tabId === -1 && thisTab.site.isFeatureEnabled('serviceworkerInitiatedRequests')`? I figure that would reduce the chance of breakage caused by some other privacy protection (e.g tracker parameter protection, smarter encryption) applying to ServiceWorker initiated requests when the `serviceworkerInitiatedRequests` feature is disabled.

* Well a line lower, so just after the `thisTab` definition in `handleRequest()`

Reply:
I'm not sure that actually makes much difference. All that happens between these two points in the code is:
1. `web_accessible_resources` check: required to protect from SWs loading our web-accessible resources.
2. `main_frame` request handling: Not possible from service workers.
3. Skip blocking for 'special' pages: prevents us from running the checks on extension + internal pages.
4. (in `blockHandleResponse`) CTL blocking: not sure if this is likely to trigger breakage, but it can be controlled via its own config.
5. Other allowlist checks: These all disable blocking in specific cases, which would therefore not introduce breakage.

The other effect of the current position is that the allowlist is fall through, meaning that, in this case, Smarter Encryption will still apply, but that will be a no-op as [mixed-content fetch from service-workers is forbidden](https://github.com/w3c/ServiceWorker/issues/493)

Note, the _advantage_ of having the block where it is now, is that it is:
 - alongside other blocking exceptions,
 - only has to be checked when the URL is a tracker URL, and
 - can therefore set the blocking metadata correctly to allow the SW exception to be shown in the dashboard and debugger.

Reply:
Alright, fair enough. Sounds good. (P.S. Don't worry about number four, I will be changing that logic very shortly.)